src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/StringCollectionTextWriter.cs

```diff
@@ -62,7 +70,19 @@ public override void Write([NotNull] char[] buffer, int index, int count)
                 throw new ArgumentOutOfRangeException(nameof(count));
             }
 
-            Content.Append(buffer, index, count);
+            while (count > 0)
+            {
+                // Split large char arrays into 1KB strings.
+                var currentCount = count;
+                if (MaxCharToStringLength < currentCount)
+                {
+                    currentCount = MaxCharToStringLength;
+                }
+
+                _content.Append(new string(buffer, index, currentCount));
+                index += currentCount;
+                count -= currentCount;
+            }
```

Comment:
Messy to duplicate `BufferedHtmlContent` code here as well as in `StringCollectionTextWriterContent`. Would it be more straightforward to make `BufferedHtmlContent.WriteTo()` `virtual` and use a `private` subclass here -- overriding `WriteTo()` and `ToString()`? Don't think `StringCollectionTextWriterContent` (the proposed subclass then) would be nearly as large.